src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -100,6 +103,13 @@ static uint16_t               low_fd_count;
 /* the list of low fd reserved */
 static int                    low_fd_list[DAOS_MIN_FD];
 
+/* default dir cache time-out in seconds*/
+#define DCACHE_TIMEOUT 10
```

Comment:
- [ ] Change default timeout

Reply:
You should either use the value in the `dfuse-dentry-dir-time` contaioner attribute or create a new `pil4dfs-dentry-dir-time`

https://github.com/daos-stack/daos/blob/master/docs/user/filesystem.md#caching

This code is going to need to handle multiple containers shortly so the code will need to handle different paths using different values here but that can be added later.

Reply:
This is exactly the same as the dfuse caching.  It's a trade-off between performance and correctness, and the optimum value will depend on what the application is doing.

There are obviously subtleties and requirement different here so having a different setting is wise but perhaps checking a pil4dfs setting and then backing off to a dfuse setting would be acceptable.

Reply:
i should have been more clear. I was not talking about functionality. i meant dfuse in this case is not used so it should not use the same value.
Since this is a user space thing, it would be better IMO to allow users to set this with an env variable (with 0 meaning dcache is off/not used) and not having to set an attr on a container. 

my concern with using the attr setting is that users almost never use or set them in practice because they are not that intuitive; or one can be using a default container that is pre created and they might not have permissions to set attrs on the container. We can try and set the default to what dfuse sees if that is provided in the ioctl of the container connect. but adding another attr on the container that users should set before is not a great option IMO.

Reply:
Users are going to do what users are going to do.  What would be good would be to have the option to set on a per-container basis for power users who will use the flexibility, there's nothing wrong with also having a environment variable as well which as you say is convenient and will cover most users I expect.

Probably the most important thing (other than to have a way to set it in the first place) is sensible default, and I absolutely agree that 60 seconds feels about right.

Reply:
I have removed the timeout of the dcache entry from this PR (not yet pushed).
However, the default timeout value (from 10s to 60s) has been already fixed in the follow-up PR.
https://github.com/daos-stack/daos/blob/84ff78657bda9589886d4c0cdefc5682b357737e/src/client/dfuse/pil4dfs/int_dfs.c#L105

Reply:
Fixed with commit b530789d44f8fded639bcf685b2e3621bf9b24b0